src/pandas_profiling/__init__.py

```diff
@@ -4,48 +4,107 @@
 """
 import json
 import warnings
-from pathlib import Path
 from datetime import datetime
+from pathlib import Path
+from typing import Union
 
-import pandas as pd
+import joblib
 import numpy as np
-from tqdm.auto import tqdm
+import pandas as pd
 
-from pandas_profiling.model.messages import MessageType
-from pandas_profiling.version import __version__
-from pandas_profiling.utils.dataframe import rename_index
-from pandas_profiling.utils.paths import get_config_default, get_config_minimal
-from pandas_profiling.config import config
+from pandas_profiling.config import config, Config
 from pandas_profiling.controller import pandas_decorator
 from pandas_profiling.model.describe import describe as describe_df
 from pandas_profiling.model.messages import MessageType
 from pandas_profiling.report import get_report_structure
+from pandas_profiling.report.presentation.abstract.renderable import Renderable
+from pandas_profiling.report.presentation.flavours.html import HTMLSequence
+from pandas_profiling.utils.dataframe import rename_index
+from pandas_profiling.utils.paths import get_config_default, get_config_minimal
+from pandas_profiling.version import __version__
 
 
 class ProfileReport(object):
     """Generate a profile report from a Dataset stored as a pandas `DataFrame`.
-    
+
     Used has is it will output its content as an HTML report in a Jupyter notebook.
     """
 
     html = ""
     """the HTML representation of the report, without the wrapper (containing `<head>` etc.)"""
 
-    def __init__(self, df, minimal=False, config_file: Path = None, **kwargs):
+    def __init__(
+        self,
+        df=None,
+        minimal=False,
+        config_file: Union[Path, str] = None,
+        lazy=True,
+        **kwargs,
+    ):
         if config_file is not None and minimal:
             raise ValueError(
                 "Arguments `config_file` and `minimal` are mutually exclusive."
             )
 
-        if minimal:
-            config_file = get_config_minimal()
+        if df is None and not lazy:
+            raise ValueError("Can init a not-lazy ProfileReport with no DataFrame")
+
+        if not config_file and not minimal and not config.is_default:
+            warnings.warn(
+                "The configuration of pandas profiling is universal, which will affect all ProfileReport "
```

Comment:
What is the function of this warning?

Reply:
If users instantiates two `ProfileReport` object in one process, they will share the config. This warning warns when user did not specify config, but `ProfileReport` implicitly used previous non-default config.